src/main/resources/assets/js/application/recruiter/services/tdprRecruiterSlotsService.js

```diff
@@ -0,0 +1,30 @@
+define(['angular', 'application/recruiter/tdprRecruiterModule', 'application/recruiter/services/tdprDateService', 'notification'], function (angular, tdprRecruiterModule) {
+    tdprRecruiterModule.service('tdprRecruiterSlotsService', ['$http', '$q', 'tdprDateService', 'dateFilter', 'AvailabilityEnum', function ($http, $q, tdprDateService, dateFilter, AvailabilityEnum) {
+
+        this.updateSlots = function (slots, personId, startDate, endDate) {
+            return $http.put("/api/slots/update/" + personId + "/" + dateFilter(startDate, "dd-MM-yyyy") + "/" + dateFilter(endDate, "dd-MM-yyyy"), this.reformatSlots(slots, personId)).then(function (response) {
+                return response;
+            }, function (err) {
+                err.message = "Failed to update slots for person. Could not change past days.";
+                return $q.reject(err);
+            });
+        };
+
+        this.filterSlots = function (slots, day) {
+            var compareData = dateFilter(day, "yyyy-MM-dd");
+            return _.filter(slots, {"day": compareData});
+        };
+
+        this.reformatSlots = function (slots, personId) {
+            return _.map(slots,
```

Comment:
It is used in updateSlots method
```
$http.put("/api/slots/update/" + personId + "/" + dateFilter(startDate, "dd-MM-yyyy") + "/" + dateFilter(endDate, "dd-MM-yyyy"), this.reformatSlots(slots, personId))
```
It is required as long as we don't have other serialising/deserialising methods for updating slots on backend.